### PR TITLE
Added IOS XR support for tacacs_server

### PR DESCRIPTION
**Pull request by jonnytpuppet**
Needs https://github.com/cisco/cisco-network-node-utils/pull/307/files

### CODE REVIEW

**Comment by jonnytpuppet** [comment]
```
tests/beaker_tests/tacacs_server/tacacs_server_provider_defaults.rb passed in 93.48 seconds
      Test Suite: tests @ 2016-03-22 17:03:33 +0000

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 93.48 seconds
      Average Test Time: 93.48 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -
```

**Comment by chburket** [comment]
:+1:

File: lib/puppet/provider/tacacs_server/cisco.rb [hunk 1]
```diff
@@ -133,9 +141,19 @@ def flush
...
+      if @property_hash.empty? ||
+         (Facter.value('operatingsystem').eql?('ios_xr') &&
+          @resource[:port] != @tacacs_server.port.to_i)
+
         # create a new Tacacs Server
-        @tacacs_server = Cisco::TacacsServerHost.new(@resource[:name])
+        create_new
+      end
+
+      if Facter.value('operatingsystem').eql?('ios_xr')
```
Comment (chrisvanheuveln) [inline lib/puppet/provider/tacacs_server/cisco.rb]: Facter calls are kind of expensive (painfully slow) on NX devices. Would it make sense to get this during `initialize` and cache it there in a class var?

**Comment by chrisvanheuveln** [comment]
Please update CHANGELOG.md.

File: lib/puppet/provider/tacacs_server/cisco.rb (hunk 1 above)
Comment (jonnytpuppet) [inline lib/puppet/provider/tacacs_server/cisco.rb]: @chrisvanheuveln the Facter ruby library already caches Factor calls. Thanks

Comment (chrisvanheuveln) [inline lib/puppet/provider/tacacs_server/cisco.rb]: Good. I thought it might.

**Comment by chrisvanheuveln** [comment]
:+1:

**Pull request merged**